workflow-templates/im-deploy-az-app-manually.yml

```diff
@@ -103,27 +110,71 @@ jobs:
           scope: ${{ env.ENVIRONMENT }}
           create-output-variables: true
         env:
-          # This variable is used to swap slots and annotate app insights
-          RESOURCE_GROUP@dev: ''
-          RESOURCE_GROUP@qa: ''
-          RESOURCE_GROUP@stage: ''
-          RESOURCE_GROUP@demo: ''
-          RESOURCE_GROUP@uat: ''
-          RESOURCE_GROUP@prod: ''
+          # Resource group holding the resources to be deployed to
+          TARGET_RESOURCE_GROUP@dev: ''
+          TARGET_RESOURCE_GROUP@qa: ''
+          TARGET_RESOURCE_GROUP@stage: ''
+          TARGET_RESOURCE_GROUP@stage-secondary: ''
+          TARGET_RESOURCE_GROUP@demo: ''
+          TARGET_RESOURCE_GROUP@uat: ''
+          TARGET_RESOURCE_GROUP@prod: ''
+          TARGET_RESOURCE_GROUP@prod-secondary: ''
+
+          # Resource group holding the state storage account and managed service identities
+          PRIMARY_RESOURCE_GROUP@dev: ''
+          PRIMARY_RESOURCE_GROUP@qa: ''
+          PRIMARY_RESOURCE_GROUP@stage stage-secondary: ''
+          PRIMARY_RESOURCE_GROUP@demo: ''
+          PRIMARY_RESOURCE_GROUP@prod prod-secondary: ''
```

Comment:
Including these items seems like kind of an advanced scenario that introduces a lot of extra work in this workflow.  Do we think this is something a lot of others use or is it specific to your team?

At least as far as the comments go for both blocks, I think most teams have the state in their "target_resource_group".

Reply:
When the development/production pipeline creates resource groups, it creates only a single storage account (for holding tfstate) in the primary region's resource group and it makes that storage account GRS (or GZRS), so that we can explicitly fail it over to the paired region, making it available again at its original URL, but now in the paired region.

So, actually most teams have their state in their primary resource group.

Reply:
See I'm confused about what the difference is for each of these resource groups.  Maybe some different naming would help clear things up?

Reply:
So it relates to the primary and secondary methodology for stage and prod.  There is a primary environment NA26 and a failover environment na27.  The target is the one you are deploying to.  The Primary is the one that has the geo redundant resources such as the storage account.  I originally had `RESOURCE_GROUP` and `PRIMARY_RESOURCE_GROUP` in this pull request.   I think the comments above explain what they are for.

Reply:
I have added little more description to the `TARGET_RESOURCE_GROUP` comment @danielle-casella-adams